plugins/destination/plugin.go

```diff
@@ -303,3 +305,24 @@ func setCqIDColumnOptionsForTables(tables []*schema.Table) {
 		setCqIDColumnOptionsForTables(table.Relations)
 	}
 }
+
+func (p *Plugin) setPKsForTables(tables schema.Tables) {
+	if p.spec.PKMode == specs.PKModeCQID {
+		setCQIDAsPrimaryKeysForTables(tables)
+	}
+}
+func setCQIDAsPrimaryKeysForTables(tables schema.Tables) {
+	for _, table := range tables {
+		for i, col := range table.Columns {
+			switch col.Name {
+			case schema.CqIDColumn.Name:
+				table.Columns[i].CreationOptions.PrimaryKey = true
+			case schema.CqParentIDColumn.Name:
+				table.Columns[i].CreationOptions.PrimaryKey = table.Parent != nil
```

Comment:
Why a parent should ever be a primary key ?

Reply:
2 different top level resources could resolve the same nested resource. In that case, the `cq_id` would be the same which means that one of the resources would be dropped

Reply:
I have removed this because the `table.Parent` is always nil